cmd/utils.go

```diff
@@ -249,32 +279,46 @@ func startProfiler(profilerType string) (minioProfiler, error) {
 			defer os.RemoveAll(dirPath)
 			return ioutil.ReadFile(fn)
 		}
-	case "mem":
-		old := runtime.MemProfileRate
-		runtime.MemProfileRate = 4096
+	case madmin.ProfilerMEM:
+		runtime.GC()
+		var buf bytes.Buffer
+		err := pprof.Lookup("heap").WriteTo(&buf, 0)
+		if err != nil {
```

Comment:
Shouldn't this be `prof.recordBase("heap")`

Reply:
Yeah. Much nicer.